src/ixdat/techniques/ec_ms.py

```diff
@@ -426,6 +429,156 @@ def siq_ecms_calibration_curve(
         else:
             return cal
 
+    # DECONVOLUTION METHODS BELOW
+    def grab_deconvoluted_current(
+        self, mol, impulse_response, tspan=None, tspan_bg=None, snr=10
+    ):
+        """Return the deconvoluted partial current for a given signal using the
+        algorithm developed by Krempl et al.
+        https://pubs.acs.org/doi/abs/10.1021/acs.analchem.1c00110
+
+        Note, this actually doesnt need the EC data - it is calculated
+        from calibrated MS data. It is only meaningful for ECMS measurements
+        at the moment, justifying placement here.
+
+        Args:
+            mol (str): Name of molecule for which deconvolution is to
+                be carried out.
+            impulse_response (ECMSImpulseResponse): Kernel object which contains the mass
+                transport parameters
+            tspan (list): Timespan for which the partial current is returned.
+            tspan_bg (list): Timespan that corresponds to the background signal.
+            snr (int): signal-to-noise ratio used for Wiener deconvolution.
+        """
+        # TODO: comments in this method so someone can tell what's going on!
+
+        # grab the calibrated data
+        t_sig, v_sig = self.grab_flux(mol, tspan=tspan, tspan_bg=tspan_bg)
+        # calculate the impulse response
+        signal_response = impulse_response.model_impulse_response_from_params(
+            dt=t_sig[1] - t_sig[0], duration=t_sig[-1] - t_sig[0]
+        )
+        # this seems quite inefficient, to re-calculate the impulse response every time?
+        # TODO: store this somehow
+        # make sure signal_response and v_sig are same length for the steps below by
+        # adding zeros to the end of whichever array is shorter
+        if len(v_sig) >= len(signal_response[1]):
+            kernel = np.hstack(
+                (signal_response[1], np.zeros(len(v_sig) - len(signal_response[1])))
+            )
+            v_sig_corr = v_sig
+        else:
+            kernel = signal_response[1]
+            v_sig_corr = np.hstack(
+                (v_sig, np.zeros(len(signal_response[1]) - len(v_sig)))
+            )
+        H = fft(kernel)
+        # TODO: store this as well.
+        partial_current = np.real(
+            ifft(fft(v_sig_corr) * np.conj(H) / (H * np.conj(H) + (1 / snr) ** 2))
+        )
+        partial_current = partial_current * sum(kernel)  # what does this do????
+        if len(t_sig) < len(partial_current):
+            delta = len(partial_current) - len(t_sig)
+            partial_current = partial_current[:-delta]
+        return t_sig, partial_current
+
+    def extract_impulse_response(self, mol, tspan=None, tspan_bg=None):
```

Comment:
the calculator constructor method
`ECMSImpulseResponse.from_data(measurement)` 
will be the "one true way" to do this, so this method can be deleted here.

Reply:
This should be deleted now!

Reply:
I believe I have deleted this. At least that's the case in my version. Let me know if incorrect.

Reply:
Ahh, now I get it - the justification for this method was as a placeholder for a method which could automatically find the t_spans. I will not write that before it's been turned into a `Calculator` though, so I deleted it.